contracts/FeeCustody.sol

```diff
@@ -0,0 +1,347 @@
+      feeDistributor.burn(address(distributionToken), toDistribute);
+    }
+
+    /**
+     * @notice
+     * Amount of _asset allocated to RBN lockers from current balance
+     * @return amount allocated to RBN lockers
+     */
+    function claimableByRBNLockersOfAsset(address _asset) external view returns (uint256) {
+      uint256 allocPCT = pctAllocationForRBNLockers;
+      return IERC20(_asset).balanceOf(address(this)).mul(allocPCT).div(TOTAL_PCT);
+    }
+
+    /**
+     * @notice
+     * Amount of _asset allocated to multisig from current balance
+     * @return amount allocated to multisig
+     */
+    function claimableByProtocolOfAsset(address _asset) external view returns (uint256) {
+      uint256 allocPCT = TOTAL_PCT.sub(pctAllocationForRBNLockers);
+      return IERC20(_asset).balanceOf(address(this)).mul(allocPCT).div(TOTAL_PCT);
+    }
+
+    /**
+     * @notice
+     * Total allocated to RBN lockers across all assets balances
+     * @return total allocated (in USD)
+     */
+    function totalClaimableByRBNLockersInUSD() external view returns (uint256) {
+      uint256 allocPCT = pctAllocationForRBNLockers;
+      return _getSwapQuote(allocPCT);
+    }
+
+    /**
+     * @notice
+     * Total allocated to multisig across all assets balances
+     * @return total allocated (in USD)
+     */
+    function totalClaimableByProtocolInUSD() external view returns (uint256) {
+      uint256 allocPCT = TOTAL_PCT.sub(pctAllocationForRBNLockers);
+      return _getSwapQuote(allocPCT);
+    }
+
+    /**
+     * @notice
+     * Total claimable across all asset balances based on allocation PCT
+     * @param _allocPCT allocation percentage
+     * @return total claimable (in USD)
+     */
+    function _getSwapQuote(uint256 _allocPCT) internal view returns (uint256 claimable) {
+      for(uint i; i < lastAssetIdx; i++){
+        IChainlink oracle = IChainlink(oracles[assets[i]]);
+
+        // Approximate claimable by multiplying
+        // current asset balance with current asset price in USD
+        claimable += IERC20(assets[i]).balanceOf(address(this))
+          .mul(oracle.latestAnswer())
+          .mul(_allocPCT)
+          .div(oracle.decimals())
+          .div(TOTAL_PCT);
+      }
+    }
+
+    /**
+     * @notice
+     * Swaps _amountIn of _asset into distributionToken
+     * @param _asset asset to swap from
+     * @param _amountIn amount to swap of asset
+     * @param _minAmountOut min amount out for every asset type swap
+     * @param _deadline deadline for transaction expiry
+     */
+    function _swap(address _asset, uint256 _amountIn, uint256 _minAmountOut, uint256 _deadline) internal {
+      TransferHelper.safeApprove(asset, address(UNIV3_SWAP_ROUTER), _amountIn);
+
+      ISwapRouter.ExactInputParams memory params =
+          ISwapRouter.ExactInputParams({
+              path: intermediaryPath[_asset],
+              recipient: msg.sender,
```

Comment:
Should the recipient be the `msg.sender`? That would send the received tokens from the swap to the caller of `distributeProtocolRevenue`

Reply:
yeah we want to send it back so that we can do one big burn() call at the [end](https://github.com/ribbon-finance/ribbonomics/pull/10/files/c072d283d9793554d8764f7b8218019a63cfb0f2#diff-7e52182ce4dccf0476db2462a3a17466b8fec9138e465c4116695ceded7e8982R126)